shell/src/main/java/alluxio/cli/fs/command/LoadCommand.java

```diff
@@ -138,4 +207,140 @@ public String getDescription() {
   public void validateArgs(CommandLine cl) throws InvalidArgumentException {
     CommandUtils.checkNumOfArgsNoLessThan(this, cl, 1);
   }
+
+  private class LoadTask implements Callable<Boolean> {
+    private final long mBlockId;
+    private final WorkerNetAddress mDataSource;
+    private final boolean mLocal;
+    private final URIStatus mStatus;
+    private final OpenFilePOptions mOptions;
+
+    LoadTask(long blockId, WorkerNetAddress dataSource, boolean local, URIStatus status,
+        OpenFilePOptions options) {
+      mBlockId = blockId;
+      mDataSource = dataSource;
+      mLocal = local;
+      mStatus = status;
+      mOptions = options;
+    }
+
+    @Override
+    public Boolean call() throws Exception {
+      BlockInfo info = mStatus.getBlockInfo(mBlockId);
+      long blockLength = info.getLength();
+      CacheRequest request = CacheRequest.newBuilder().setBlockId(mBlockId).setLength(blockLength)
+          .setOpenUfsBlockOptions(getOpenUfsBlockOptions(mBlockId, mStatus, mOptions))
+          .setSourceHost(mDataSource.getHost()).setSourcePort(mDataSource.getDataPort()).build();
+      // TODO(jianjian) support FUSE processLocalWorker?
+      WorkerNetAddress worker;
+      try {
+        if (mLocal) {
+          // send request to local worker
+          worker = mFsContext.getNodeLocalWorker();
+        } else { // send request to data source
+          worker = mDataSource;
+        }
+        try (CloseableResource<BlockWorkerClient> blockWorker =
+            mFsContext.acquireBlockWorkerClient(worker)) {
+          CacheResponse response = blockWorker.get().cache(request);
+          return response.getSuccess();
+        }
+      } catch (Exception e) {
+        System.out.printf("Failed to complete cache request for block %d of file %s: %s", mBlockId,
+            mStatus.getPath(), e.toString());
+      }
+      return false;
+    }
+  }
+
+  // todo(jianjian) consider refactoring with AlluxioBlockStore
+  private WorkerNetAddress getDataSouce(long blockId, URIStatus status, BlockLocationPolicy policy)
```

Comment:
this is pretty long and complicated. better to be consolidated with `alluxio.client.block.AlluxioBlockStore#getInStream()`
e.g., making a util method called both by here and `AlluxioBlockStore#getInStream()`

Reply:
refactor AlluxioBlockStore to expose a method called getDataSourceAndType instead of creating a util, because blockStore are suppose to provide block level operations.